server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -640,10 +641,22 @@
         String resourceType = cmd.getResourceType();
         String customerName = cmd.getCustomer();
         boolean listAll = cmd.listAll();
+        Long projectId = cmd.getProjectId();
+
+        if (projectId == null && ResourceObjectType.Project.name().equalsIgnoreCase(resourceType) && !Strings.isNullOrEmpty(resourceId)) {
+            try {
+                projectId = Long.parseLong(resourceId);
+            } catch (final NumberFormatException e) {
+                final ProjectVO project = _projectDao.findByUuidIncludingRemoved(resourceId);
+                if (project != null) {
+                    projectId = project.getId();
+                }
```

Comment:
return empty if projectId is null ?

Reply:
Yes, if projectid is not passed and invalid resourceid is passed.

Reply:
so, if projectId is still null here, return empty result or continue on searching ?

Reply:
This will continue to search and return empty result/response. This only happens when a user does not pass the projectId and invalid resourceid or does not pass the resourceid. I checked against 4.11 for behaviour and tried to fix in a way that user experience is same as 4.11.

Reply:
sounds good to me.